pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -218,6 +218,27 @@ runCmd() {
     "$@"
 }
 
+getNixpkgsPath() {
+    nixpkgsPath=
+
+    # Get NixOS configuration
+    NIXOS_CONFIG=${NIXOS_CONFIG:-$(runCmd nix-instantiate --find-file nixos-config)}
+    if [[ -d $NIXOS_CONFIG ]]; then
+        NIXOS_CONFIG=$NIXOS_CONFIG/default.nix
+    fi
+
+    # Try config.nixpkgs.source from build file
+    if [[ -z $buildingAttribute ]]; then
+        if nixpkgsPath=$(runCmd nix-instantiate "$buildFile" -A "${attr:+$attr.}config.nixpkgs.source" 2>/dev/null | jq -r); then return; fi
+    fi
+
+    # Try config.nixpkgs.source from NixOS config
+    if nixpkgsPath=$(runCmd nix-instantiate --eval --expr "((import <nixpkgs> {}).callPackage $NIXOS_CONFIG {}).nixpkgs.source" 2>/dev/null | jq -r); then return; fi
```

Comment:
`pkgs.callPackage` of NixOS configuration module/file is not how you evaluate NixOS configuration. if someone has `nixpkgs.source` defined in different file, or under `config` attrname, this will fail.

```suggestion
    if nixpkgsPath=$(runCmd nix-instantiate --eval --expr "((import <nixpkgs/nixos/eval-config.nix> { modules = [ '$NIXOS_CONFIG' ]; }).config.nixpkgs.source" 2>/dev/null | jq -r); then return; fi
```
or
```suggestion
    if nixpkgsPath=$(runCmd nix-instantiate --eval --expr "((import <nixpkgs> { config = {}; overlays = []; }).nixos (import '$NIXOS_CONFIG')).config.nixpkgs.source" 2>/dev/null | jq -r); then return; fi
```

Reply:
Eh, you found the biggest hack of my proposal. Evaluating the whole NixOS configuration (with whatever `<nixpkgs>` is currently pointing to) is not generally possible: there may be errors due to newer options being used, or older being removed.

The only safe way is to use `pkgs.callPackage` to fill in whatever arguments the configuration.nix wants and evaluating *only* `nixpkgs.source` from the resuling attrset. This works as longs as users stick to the prescriptions I wrote in nixpkgs.source.description:

> In order to be evaluated before building the system, this option must be set in your configuration entry point (i.e. configuration.nix) and the value must not reference any of the NixOS module arguments (i.e. `config`, `pkgs`, etc.).

Reply:
I was wrong, I've found out about `_module.check = false` and for what I've tested so far it seems to be working. I can evaluate a configuration with errors and extract the `config.nixpkgs.source`.